src/Groups/GAPGroups.jl

```diff
@@ -0,0 +1,617 @@
+    cc = x ^ G
+where G is a group and x = `representative`(`cc`) is either an element or a subgroup of G.
+"""
+struct GroupConjClass{T<:Group, S<:Union{GAPGroupElem,Group}}
+   X::T
+   repr::S
+   CC::GapObj
+end
+
+Base.show(io::IO, x::GroupConjClass) = print(io, GAP.gap_to_julia(GAP.Globals.StringView(x.repr))*" ^ "*GAP.gap_to_julia(GAP.Globals.StringView(x.X)))
+
+function _conjugacy_class(G, g, cc::GapObj)         # function for assignment
+  return GroupConjClass{typeof(G), typeof(g)}(G, g, cc)
+end
+
+==(a::GroupConjClass{T, S}, b::GroupConjClass{T, S}) where S where T = a.CC == b.CC 
+
+order(C::GroupConjClass) = GAP.Globals.Size(C.CC)
+
+representative(C::GroupConjClass) = C.repr
+
+number_conjugacy_classes(G::Group) = GAP.Globals.NrConjugacyClasses(G.X)
+
+# START elements conjugation
+
+"""
+    conjugacy_class(G::Group, g::GAPGroupElem) -> GroupConjClass
+return the conjugacy class `cc` of `g` in `G`, where `g` = `representative`(`cc`).
+"""
+function conjugacy_class(G::Group, g::GAPGroupElem)
+   return _conjugacy_class(G, g, GAP.Globals.ConjugacyClass(G.X,g.X))
+end
+
+function Base.rand(C::GroupConjClass{S,T}) where S where T<:GAPGroupElem
+   return group_element(C.X, GAP.Globals.Random(C.CC))
+end
+
+"""
+    elements(C::GroupConjClass)
+return the array of the elements in C.
+"""
+function elements(C::GroupConjClass{S, T}) where S where T<:GAPGroupElem
+   L=GAP.Globals.AsList(C.CC)
+   l = Vector{T}(undef, length(L))
+   for i in 1:length(l)
+      l[i] = group_element(C.X,L[i])
+   end
+   return l
+end
+
+"""
+    conjugacy_classes(G::Group)
+returns the array of all conjugacy classes of elements in G. It is guaranteed that the class of the identity is in the first position.
+"""
+function conjugacy_classes(G::Group)
+   L=GAP.gap_to_julia(GAP.Globals.ConjugacyClasses(G.X))
+   return GroupConjClass{typeof(G), elem_type(G)}[ _conjugacy_class(G,group_element(G,GAP.Globals.Representative(cc)),cc) for cc in L]
+end
+
+Base.:^(x::GAPGroupElem, y::GAPGroupElem) = group_element(_maxgroup(parent(x), parent(y)), x.X ^ y.X)
+
+"""
+    isconjugate(G::Group, x::GAPGroupElem, y::GAPGroupElem)
+if `x`,`y` are conjugate in `G`, return 
+```
+true, z
+```
+where `x^z=y`; otherwise, return
+```
+false, nothing
+```
+"""
+function isconjugate(G::Group, x::GAPGroupElem, y::GAPGroupElem)
+   if GAP.Globals.IsConjugate(G.X, x.X, y.X)
+      return true, group_element(G,GAP.Globals.RepresentativeAction(G.X, x.X, y.X))
+   else
+      return false, nothing
+   end
+end
+# END elements conjugation 
+
+# START subgroups conjugation
+"""
+    conjugacy_class(G::T, H::T) where T<:Group -> GroupConjClass
+return the subgroup conjugacy class `cc` of `H` in `G`, where `H` = `representative`(`cc`).
+"""
+function conjugacy_class(G::T, g::T) where T<:Group
+   return _conjugacy_class(G, g, GAP.Globals.ConjugacyClassSubgroups(G.X,g.X))
+end
+
+function Base.rand(C::GroupConjClass{S,T}) where S where T<:Group
```

Comment:
You can follow on how `AbstractAlgebra.rand(::Perm)` is implemented to have all of the goodies of `Base.rand`

Reply:
I didn't comment on that because this is probably not the highest priority, but I know well the `Random` module, so if I can help, ask me any question!

Reply:
indeed You know it definitely better than I do ;-)

Reply:
While I don't know Julia's `Random` module at all, I know random in GAP quite well; one of the things one wants to support there is allowing to optionally pass a "random state" / `RNG` to be used by the `Random` function. A quick glance at https://docs.julialang.org/en/v1/stdlib/Random/ leads me to believe that this is not so different from what Julia does (GAP even uses a Mersenne twister as default. Well, almost -- it's complicated... but let's ignore that). But of course the devil is in the details...

Anyway, for "basic users" this is not so important, but for e.g. group recognition (= my SFB project) it certainly is. Doing that shouldn't be so hard, but of course one can now wonder how to expose this to the  user:
- one could provide a Julia wrapper type for GAP RNGs;
- one could provide a GAP wrapper type for Julia RNgs;
- or even both...
- one could even think about modifying GAP.jl to instruct GAP to use Julia RNGs by default

Reply:
In this case I think GAP should use julias RNG. otherwise we need to warn users to use `GAPseed!(...)` when they want to get a deterministic result;  But usage of GAP deep in some Oscar function is implementation detail (users don't have to know that the underlying functions call some randomized versions of algorithms in GAP) so users will not know that they need to use it.